internal/services/dynatrace/dynatrace_tag_rules_resource.go

```diff
@@ -0,0 +1,273 @@
+package dynatrace
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/hashicorp/go-azure-helpers/lang/response"
+	"github.com/hashicorp/go-azure-sdk/resource-manager/dynatrace/2023-04-27/monitors"
+	"github.com/hashicorp/go-azure-sdk/resource-manager/dynatrace/2023-04-27/tagrules"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/sdk"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/pluginsdk"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/validation"
+)
+
+type TagRulesResource struct{}
+
+type TagRulesResourceModel struct {
+	Name        string       `tfschema:"name"`
+	Monitor     string       `tfschema:"monitor_id"`
+	LogRules    []LogRule    `tfschema:"log_rule"`
+	MetricRules []MetricRule `tfschema:"metric_rule"`
+}
+
+type MetricRule struct {
+	FilteringTags []FilteringTag `tfschema:"filtering_tag"`
+}
+
+type LogRule struct {
+	FilteringTags        []FilteringTag `tfschema:"filtering_tag"`
+	SendAadLogs          bool           `tfschema:"send_azure_active_directory_logs_enabled"`
+	SendActivityLogs     bool           `tfschema:"send_activity_logs_enabled"`
+	SendSubscriptionLogs bool           `tfschema:"send_subscription_logs_enabled"`
+}
+
+type FilteringTag struct {
+	Name   string `tfschema:"name"`
+	Value  string `tfschema:"value"`
+	Action string `tfschema:"action"`
+}
+
+func (r TagRulesResource) Arguments() map[string]*schema.Schema {
+	return map[string]*schema.Schema{
+		"name": {
+			Type:         pluginsdk.TypeString,
+			Required:     true,
+			ForceNew:     true,
+			ValidateFunc: validation.StringIsNotEmpty,
+		},
+
+		"monitor_id": {
+			Type:         pluginsdk.TypeString,
+			Required:     true,
+			ForceNew:     true,
+			ValidateFunc: monitors.ValidateMonitorID,
+		},
+
+		"log_rule": {
+			Type:     pluginsdk.TypeList,
+			Optional: true,
+			ForceNew: true,
+			MaxItems: 1,
+			Elem: &pluginsdk.Resource{
+				Schema: map[string]*schema.Schema{
+					"send_azure_active_directory_logs_enabled": {
+						Type:     pluginsdk.TypeBool,
+						Optional: true,
+					},
+
+					"send_activity_logs_enabled": {
+						Type:     pluginsdk.TypeBool,
+						Optional: true,
+					},
+
+					"send_subscription_logs_enabled": {
+						Type:     pluginsdk.TypeBool,
+						Optional: true,
+					},
```

Comment:
do these have default values?

Reply:
Yes, the default value is `false`. I'll update it.